Unbound.xs

```diff
@@ -92,15 +93,24 @@ static bool _decrement_dub_ctx_refcount (pTHX_ DNS__Unbound__Context* dub_ctx) {
     if (!--dub_ctx->refcount) {
         _DEBUG("Freeing DNS__Unbound__Context %p", dub_ctx);
 
-        if ((getpid() == dub_ctx->pid) && PL_dirty) {
-            warn("Freeing DNS::Unbound context at global destruction; memory leak likely!");
-        }
+        // To avoid a hang in subprocesses, we forgo ub_ctx_delete() except
+        // in the original process. This accommodates an upstream bug that
+        // seems to affect (at least) AlmaLinux 9 and Ubuntu 22:
+        //
+        // https://github.com/NLnetLabs/unbound/issues/775
+        //
+        if (getpid() == dub_ctx->pid) {
+            if (PL_dirty) {
+                warn("Freeing DNS::Unbound context at global destruction; memory leak likely!");
+            }
 
-        // We do NOT need to _close_saved_debugfd() here because
-        // Unbound will do that for us.
+            // We do NOT need to _close_saved_debugfd() here because
+            // Unbound will do that for us.
 
-        ub_ctx_delete(dub_ctx->ub_ctx);
-        dub_ctx->ub_ctx = NULL;
+            ub_ctx_delete(dub_ctx->ub_ctx);
```

Comment:
If we stop calling `ub_ctx_delete` on != pid we are mainly skipping the fix from upstream
https://github.com/NLnetLabs/unbound/issues/775
note this would also not call `free` on the struct elements